timelock-corruption-detection/src/main/java/com/palantir/history/models/PaxosHistoryOnSingleNode.java

```diff
@@ -17,14 +17,38 @@
 package com.palantir.history.models;
 
 import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
 
 import org.immutables.value.Value;
 
+import com.google.common.collect.ImmutableMap;
 import com.palantir.paxos.NamespaceAndUseCase;
 
 @Value.Immutable
 public interface PaxosHistoryOnSingleNode {
 
     @Value.Parameter
     Map<NamespaceAndUseCase, LearnerAndAcceptorRecords> history();
+
+    default Map<Long, LearnedAndAcceptedValue> getConsolidatedLocalAndRemoteRecord(
```

Comment:
If we keep this model, it may be worth doing some kind of caching. I'm wondering if a derived or lazy `Map<NamespaceAndUseCase, Map<Long, LearnedAndAcceptedValue>` is a good idea, even